### PR TITLE
Arachne 1527

### PR DESCRIPTION
**Pull request by agackovka**

### CODE REVIEW

File: src/modules/DataCatalog/components/List/Actions/container.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ class DataCatalogListActionsBuilder {
 
   mapStateToProps(state) {
     return {
-      currentSearchStr: state.routing.locationBeforeTransitions.search,
+      currentSearchStr: state.routing.locationBeforeTransitions.query,
```
Comment (pavgra) [inline src/modules/DataCatalog/components/List/Actions/container.js]: This will not work

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "arachne-ui",
-  "version": "1.9.28",
+  "version": "1.9.29",
```
Comment (johnSamilin) [inline package.json]: 1.10.x

File: src/modules/DataCatalog/components/List/Actions/container.js [hunk 3]
```diff
@@ -32,8 +33,10 @@ class DataCatalogListActionsBuilder {
   }
 
   mapStateToProps(state) {
+    const cleanPath = get(state, 'routing.locationBeforeTransitions.pathname');
```
Comment (johnSamilin) [inline src/modules/DataCatalog/components/List/Actions/container.js]: default value

File: src/modules/StudyManager/components/List/Actions/container.js [hunk 4]
```diff
@@ -64,26 +63,17 @@ export default class StudyViewBuilder extends ContainerBuilder {
...
       ...dispatchProps,
       ...ownProps,
       refresh() {
-        const url = new Uri(stateProps.cleanPath);
-        url.setSearch(stateProps.currentQuery);
-        url.setSearch({
-          hash: Math.random().toString(36).substring(7),
-        });
-        dispatchProps.go(url.href());
+        dispatchProps.go(Utils.getHref(path, currentQuery, true));
```
Comment (pavgra) [inline src/modules/StudyManager/components/List/Actions/container.js]: use actions.router.setSearch

File: src/services/Utils.js [hunk 5]
```diff
@@ -43,6 +43,7 @@ import { ModalUtils } from 'arachne-ui-components';
 import types from 'const/modelAttributes';
 import URI from 'urijs';
 
+
```
Comment (pavgra) [inline src/services/Utils.js]: why?

File: src/modules/DataCatalog/ducks/dataSourceList.js [hunk 6]
```diff
@@ -25,7 +25,8 @@ import { apiPaths } from 'modules/DataCatalog/const';
 
 const dsList = new Duck({
   name: 'DC_DATA_SOURCE_LIST',
-  urlBuilder: ({ searchStr }) => {
+  urlBuilder: (props) => {
+    const searchStr = props.searchStr;
```
Comment (pavgra) [inline src/modules/DataCatalog/ducks/dataSourceList.js]: why?

File: src/modules/DataCatalog/components/List/Actions/container.js [hunk 7]
```diff
@@ -52,7 +55,7 @@ class DataCatalogListActionsBuilder {
       ...stateProps,
       ...dispatchProps,
       reload: () => {
-        dispatchProps.loadDsList({ searchStr: stateProps.currentSearchStr });
+        dispatchProps.setSearch({ hash: Math.random().toString(36).substring(7) })
```
Comment (pavgra) [inline src/modules/DataCatalog/components/List/Actions/container.js]: move to utils

File: src/modules/DataCatalog/components/List/Actions/container.js [hunk 8]
```diff
@@ -32,8 +32,10 @@ class DataCatalogListActionsBuilder {
   }
 
   mapStateToProps(state) {
+    const cleanPath = get(state, 'routing.locationBeforeTransitions.pathname');
     return {
-      currentSearchStr: state.routing.locationBeforeTransitions.search,
+      currentQuery: state.routing.locationBeforeTransitions.query,
```
Comment (pavgra) [inline src/modules/DataCatalog/components/List/Actions/container.js]: who uses it?

File: src/modules/DataCatalog/components/List/Actions/container.js [hunk 9]
```diff
@@ -52,7 +55,7 @@ class DataCatalogListActionsBuilder {
       ...stateProps,
       ...dispatchProps,
       reload: () => {
```
Comment (pavgra) [inline src/modules/DataCatalog/components/List/Actions/container.js]: may be, create appropriate action creater in router namespace?

File: src/modules/DataCatalog/components/List/Actions/container.js [hunk 10]
```diff
@@ -33,7 +33,7 @@ class DataCatalogListActionsBuilder {
 
   mapStateToProps(state) {
     return {
-      currentSearchStr: state.routing.locationBeforeTransitions.search,
+      cleanPath: get(state, 'routing.locationBeforeTransitions.pathname')
```
Comment (pavgra) [inline src/modules/DataCatalog/components/List/Actions/container.js]: seems like unused variable

File: src/modules/DataCatalog/components/List/Actions/container.js [hunk 11]
```diff
@@ -62,7 +52,6 @@ class DataCatalogListActionsBuilder {
       Component: this.getComponent(),
```
Comment (pavgra) [inline src/modules/DataCatalog/components/List/Actions/container.js]: use ContainerBuilder

**Pull request merged**